### PR TITLE
feat(compass-import-export): run export and show progress toasts COMPASS-6580

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-6580

#### Description
This pr updates the new export flow to run the actual export and show progress toasts. After clicking the export button we show the output file selection. We show toasts similarly to how we do it in import (pr https://github.com/mongodb-js/compass/pull/4169).  When disconnecting we abort the export.

End to end tests will be in https://jira.mongodb.org/browse/COMPASS-6731 

#### Screenshots
**in progress**
Screenshot 2023-04-19 at 9 31 06 PM

**csv download phase in progress**
Screenshot 2023-04-19 at 9 43 23 PM

**success**
Screenshot 2023-04-19 at 9 30 25 PM

**error**
Screenshot 2023-04-19 at 9 33 44 PM

**stopped**
Screenshot 2023-04-19 at 9 31 11 PM

#### Note to reviewer
- Going to check with product/design on the text we show for the download phase of a csv export. That is likely subject to change. Open to suggestions.  https://mongodb.slack.com/archives/GDTJXPHD0/p1682010712466919

### CODE REVIEW

File: packages/compass-import-export/src/modules/export.ts [hunk 1]
```diff
@@ -257,18 +323,204 @@ export const selectFieldsToExport = (): ExportThunkAction<
...
+    const query =
+      selectedFieldOption === 'select-fields'
+        ? {
+            ...(_query ?? {
+              filter: {},
+            }),
+            projection: createProjectionFromSchemaFields(
+              Object.values(fieldsToExport)
+                .filter((field) => field.selected)
+                .map((field) => field.path)
```
Comment (Anemy) [inline packages/compass-import-export/src/modules/export.ts:374]: I'll move this into a helper function we unit test.

**Comment by lerouxb** [comment]
There's still the existing flake where imports (especially on RHEL) sometimes complete before we can abort it.

**Review by lerouxb — APPROVED** [review]
Looks good to me. Did some manual testing as well and it works well. Nice work!

**Comment by Anemy** [comment]
Merged with the flaking import abort test, gonna work on fixing that flake.

**Pull request merged**